### ISSUE TITLE
Custom namespace support

### ISSUE THREAD

**Issue by jacqueslelezard** [w3c/epub-specs#960]
Hi,
I would like to be sure: are custom namespace authorized in epub 3.1 ?

For example can I add my namespace:
`<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops" xmlns:custom="http://custom-namespace.com">`
and then use my own prefixed tags:
`<custom:menu>This is the menu</custom:menu>` ?

It doesn't seems to be supported by epubcheck but I saw epub 3.1 was supposed to support custom attributes: [reference](https://www.w3.org/Submission/epub-contentdocs/#sec-xhtml-custom-attributes)

Did I misunderstood the w3 doc or is it an epubcheck bug? There is similar question [here](https://github.com/IDPF/epubcheck/issues/491) about svg.

Thanks

**Comment by mattgarrish** [comment]
The HTML specification states:

> When vendor-neutral extensions to this specification are needed, either this specification can be updated accordingly, or an extension specification can be written that overrides the requirements in this specification.

https://www.w3.org/TR/html51/infrastructure.html#conformance-requirements-extensibility

In other words, you can create extensions using new namespaces, but they are not valid HTML unless they are recognized formally as such. You can't just insert random namespaced elements and have them validate.

EPUB is itself an extension of HTML5, as that's what makes the use of epub:type and the prefixed elements valid. Validate a document using those against a strict html validator and you'll get errors.

In other words, for acceptance in EPUB, you would have to submit a proposal for your extension to be added and then epubcheck could be updated to allow. So it's working as expected.

(The likelihood of element extensions being added that aren't formally recognized by W3C are probably as near to zero as you can get these days, however.)

SVG takes a more permissive approach than HTML:

> SVG allows inclusion of elements from foreign namespaces anywhere with the SVG content. In general, the SVG user agent will include the unknown elements in the DOM but will otherwise ignore unknown elements.

https://www.w3.org/TR/SVG/extend.html#ForeignNamespaces

**Comment by jacqueslelezard** [comment]
Thanks for the detailed explanation @mattgarrish
